core/tests/puppeteer-acceptance-tests/user-utilities/exploration-creator-utils.ts

```diff
@@ -0,0 +1,649 @@
+   */
+  async expectPreviewSummaryToBeVisible(): Promise<void> {
+    await this.page.waitForSelector('.e2e-test-open-preview-summary-modal');
+    const previewSummary = await this.page.$(
+      '.e2e-test-open-preview-summary-modal'
+    );
+
+    if (previewSummary) {
+      showMessage('Preview summary is visible.');
+    } else {
+      throw new Error('Preview summary is not visible.');
+    }
+  }
+
+  /**
+   * This function helps in updating advanced settings
+   */
+  async enableAutomaticTextToSpeech(): Promise<void> {
+    await this.clickOn(textToSpeechToggle);
+  }
+
+  /**
+   * This function checks whether the Automatic Text-to-Speech
+   * setting is enabled or disabled.
+   */
+  async expectAutomaticTextToSpeechToBeEnabled(): Promise<void> {
+    await this.page.waitForSelector('#text-speech-switch');
+    const autoTTSwitch = await this.page.$('#text-speech-switch');
+    const isAutoTTSwitchOn = await this.page.evaluate(
+      switchElement => switchElement.checked,
+      autoTTSwitch
+    );
+    if (isAutoTTSwitchOn) {
+      showMessage('Automatic Text-to-Speech is enabled.');
+    } else {
+      throw error('Automatic Text-to-Speech is disabled.');
+    }
+  }
+
+  /**
+   * This function helps in assigning role of collaborator to any guest user.
+   */
+  async assignUserToCollaboratorRole(username: string): Promise<void> {
+    await this.clickOn(editbutton);
+    await this.clickOn(addUserName);
+    await this.type(addUserName, username);
+    await this.clickOn(addRoleBar);
+    await this.clickOn(collaborator);
+    await this.clickOn(saveRole);
+    showMessage(`${username} has been added as collaborator.`);
+  }
+
+  /**
+   * This function helps in assigning role of Playtester to guest user.
+   */
+  async assignUserToPlaytesterRole(username: string): Promise<void> {
+    await this.clickOn(editbutton);
+    await this.page.waitForSelector('.e2e-test-editor-role-names', {
+      visible: true,
+    });
+    await this.clickOn(addUserName);
+    await this.type(addUserName, username);
+    await this.clickOn(addRoleBar);
+    await this.clickOn(playtester);
+    await this.clickOn(saveRole);
+    showMessage(`${username} has been added as playtester.`);
+  }
+
+  /**
+   *Exception function to verify the setting
+   *of the exploration to Public/Private
+   */
+  async expectExplorationToBePublished(): Promise<void> {
+    if (this.isViewportAtMobileWidth()) {
+      await this.clickOn(publishButtonDropDown);
+      const publishButton = await this.page.$(mobilePublishButton);
+      if (!publishButton) {
+        showMessage(
+          'Exploration is set to Public and is accessible to Oppia users.'
+        );
+      } else {
+        throw new Error(
+          'Exploration is set to Private and is not' +
+            ' accessible to Oppia users.'
+        );
+      }
+    } else {
+      const publishButton = await this.page.$('.e2e-test-publish-exploration');
+      if (!publishButton) {
```

Comment:
All common code should go outside the if/else, don't repeat code unnecessarily. Minimize the amount of code that is repeated, otherwise you end up with multiple copies of the same code to maintain and it's hard to keep those in sync.

Reply:
Its not common actually, We are only using it here. In the mobile acceptance test we are using different publishButton.
This is very much specific to this area.
I think you may get confused with the name as it is used in different places outside also. So, renaming it.